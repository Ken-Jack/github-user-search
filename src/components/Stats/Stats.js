// @flow

import React from 'react';
import {connect} from 'react-fela';

const map = require('lodash/fp/map').convert({cap: false});

type Props = {
  stats: Object,
  styles: any,
};

function Stats({stats, styles}: Props) {
  return (
    <ul className={styles.root}>
      {
        map((value, key) => {
          return (
            <li
              className={styles.item}
              key={key}
            >
              <p className={styles.value}>{value}</p>
              <p className={styles.label}>{key}</p>
            </li>
          );
        }, stats)
      }
    </ul>
  );
}

const styles = {
  root: () => ({
    display: 'flex',
  }),

  item: () => ({
    marginRight: '28px',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  }),

  value: () => ({
    fontSize: '30px',
  }),

  label: () => ({
    fontSize: '14px',
  }),
};

export default connect(styles)(Stats);
