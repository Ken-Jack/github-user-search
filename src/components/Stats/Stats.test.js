import React from 'react';
import felaSnapshot from 'test-util/fela-snapshot';
import Stats from './Stats';

describe('Component: Stats', () => {

  it('should render', () => {
    const stats = {
      Foo: 'bar',
      Bla: 'yo',
    };
    const component = felaSnapshot(
      <Stats
        stats={stats}
      />
    );
    expect(component).toMatchSnapshot();
  });

});
