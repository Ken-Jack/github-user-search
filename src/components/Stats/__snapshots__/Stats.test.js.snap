// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Stats should render 1`] = `
Object {
  "component": <ul
    className="a"
  >
    <li
      className="b a c d"
    >
      <p
        className="e"
      >
        bar
      </p>
      <p
        className="f"
      >
        Foo
      </p>
    </li>
    <li
      className="b a c d"
    >
      <p
        className="e"
      >
        yo
      </p>
      <p
        className="f"
      >
        Bla
      </p>
    </li>
  </ul>,
  "styles": "
    .a{display:flex}
    .b{margin-right:28px}
    .c{flex-direction:column}
    .d{align-items:center}
    .e{font-size:30px}
    .f{font-size:14px}",
}
`;
